.github/workflows/publish.yml

```diff
@@ -1,33 +1,33 @@
 name: Publish
 
+env:
+  FORCE_COLOR: 3
+  PIP_DISABLE_PIP_VERSION_CHECK: 1
+
 on:
   push:
     tags:
       - '*v*'
 
-env:
-  FORCE_COLOR: 3
-  PIP_DISABLE_PIP_VERSION_CHECK: 1
-
 jobs:
   publish:
     name: Publish
     runs-on: ubuntu-latest
     steps:
-      - name: Check out repository
-        uses: actions/checkout@96f53100ba2a5449eb71d2e6604bbcd94b9449b5
-      - name: Set up Node.js
-        uses: actions/setup-node@698d50532ea4cc0490274810a45fe52318110a1c
+      - name: Check out repo
+        uses: actions/checkout@v3
```

Comment:
using hashes is a security recommendation. Put the version as a comment next to the hash.

Reply:
Better to address this one separately. Lots of manual work to get each individual hash. @jrandolf is there a way to automate this process?

Reply:
we could copy the hashes from this workflow (or perhaps from Puppeteer) and then let dependabot do the updates

Reply:
e.g., https://github.com/puppeteer/puppeteer/blob/main/.github/workflows/publish.yml#L19

Reply:
btw I counted only four unique packages so updating manually (find & replace) should not be too problematic

Reply:
There are way more than four, updating them is quite tedious